cmd/agent/app/servers/tbuffered_server_test.go

```diff
@@ -98,11 +98,13 @@ type fakeTransport struct {
 
 func (t *fakeTransport) Read(p []byte) (n int, err error) {
 	packet := t.packet.Inc()
-	if packet > 2 {
-		if packet > 3 {
-			// return error once when packet==3, otherwise block
-			t.wg.Wait()
-		}
+	if packet == 2 {
+		// return some error packet, followed by valid one
+		return 0, io.ErrNoProgress
+	}
+	if packet > 3 {
+		// don't return error at this moment to avoid flaky test
+		t.wg.Wait()
```

Comment:
I don't remember the exact semantics of this test, but from a cursory review, previously the case `packet == 3` would result in EOF with no wait(), while now it falls through to the range loop below. Is that expected?

Reply:
Yes, the previous behaviour was: send two packets, the third one will return EOF immediately. And the flake could be if the error wasn't still handled at the moment we expect that and there is no was to watch it. That's why I changed the order for second packet and error - having a dropped second packet guarantees that the error is already handled